### PR TITLE
Upgrade dependencies across major versions.

### LINKED ISSUE #183: Migrate the `url_launcher_extended` package to null safety

### PR DESCRIPTION
**Pull request by Jonas-Sander**
Upgrade dependencies across major versions.  
Remove unused dependencies.  

**Removed `package:firebase`**  
We used [`package:firebase`](https://pub.dev/packages/firebase) before, which was discontinued.    
This package was used to implement:
* `lib/analytics/lib/src/analytics/backend/firebase_web_analytics_backend.dart`
* `lib/filesharing/filesharing_logic/lib/src/file_uploader/implementation/web_firebase_file_uploader.dart`
* `lib/remote_configuration/lib/src/implementation/firebase_web_remote_configuration.dart`  

Since the plugin now support the Web natively I removed the mentioned files.  
From what I've seen this also means that we could probably remove our custom wrapper classes entirely and just use the native Firebase classes again in our code. I haven't done this yet, I opened #370 for this.

Fixes #183

### CODE REVIEW

**Comment by nilsreichardt** [comment]
Oh, nice work! When we have this PR, have I can implement #221 which should unblock me of finishing #314 👌

**Comment by Jonas-Sander** [comment]
> Oh, nice work! When we have this PR, have I can implement #221 which should unblock me of finishing #314 👌

I'll had to do this in some kind of form anyways, since I can't launch web with no `FirebaseOptions` passed.  

I have run:
* `flutterfire configure --out=lib\firebase_options_dev.g.dart -y -p sharezone-debug`
* `flutterfire configure --out=lib\firebase_options_prod.g.dart -y -p sharezone-c2bd8`

This generated the following files:
* `app/android/app/src/dev/google-services.json` (had to move it manually to the /src/dev folder to replace the old one)
* `app/android/app/src/prod/google-services.json` (had to move it manually to the /src/prod folder to replace the old one)
* `app/ios/firebase_app_id_file.json` (sharezone-c2bd8, active)
* `app/ios/firebase_app_id_file_dev.json` (sharezone-debug, inactive)
* `app/lib/firebase_options_prod.g.dart`
* `app/lib/firebase_options_dev.g.dart` (sharezone-debug, active (Web))
* `app/macos/firebase_app_id_file.json` (sharezone-c2bd8, active)
* `app/macos/firebase_app_id_file_dev.json` (sharezone-debug, inactive)

"active" means in use by default.

So for the `macos/firebase_app_id_file.json` and `ios/firebase_app_id_file.json` files there is only one active one at most. There are no different folders for the different flavors like for the `google-services.json` files on Android. They [seem to be only used for Crashlytics](https://github.com/firebase/flutterfire/discussions/8691#discussioncomment-3919740) so I used the `sharezone-c2bd8`  as the active ones. 
I'm pretty sure there is a better way to do this so that it works with flavors but I don't know how.

Also this will break the script that replaces the dev credentials on web (since it works via the `<script src="firebase_init_debug.js"></script>` imports. Since this PR is very large already I would probably say that this should be fixed in another PR. WDYT?

**Comment by Jonas-Sander** [comment]
Could I maybe just pass the rest of the firebase + flavor work onto you @nilsreichardt ? (After this is merged)

**Comment by nilsreichardt** [comment]
Yes, of course 👍

File: lib/sharezone_widgets/lib/src/svg.dart
Comment (Jonas-Sander) [inline lib/sharezone_widgets/lib/src/svg.dart:27]: Why is the `PlatformSvg` class even needed if we just forward to `SvgPicture` anyways? Couldn't we just remove `PlatformSvg`? @nilsreichardt  

(Should I do it?)

**Comment by Sharezone-Admin** [comment]
⬇️ Generated builds by [Codemagic](https://codemagic.io/app/629c82ea463af7ff553fc7a5/build/63f7b0617089b1a95e44a0d2) for commit bdc95747e477c952ee728c7130cd346bf7640592 ⬇️

_Note_: Only Sharezone team members are able to install the iOS app.

| Android |
|:-:|
| image <br /> [Download link](https://api.codemagic.io/artifacts/42166b08-cbca-497c-98e8-52a4f7857413/de980200-586e-49c5-a609-ae96003f8ebc/app-dev-release.apk) |

**Comment by Jonas-Sander** [comment]
iOS build failed
```

Error: CocoaPods's specs repository is too out-of-date to satisfy dependencies.
To update the CocoaPods specs, run:
  pod repo update

Error running pod install


Build failed :|
Step 7 script `Build IPA` exited with status code 1
```

Either you have to do that for me or I'll do it maybe tomorrow.  
I'll have my MacBook burried somewhere right now :D

**Comment by nilsreichardt** [comment]
> Also we should thoroughly test the app since many different packages had new major versions (I just updated them all automatically and fixed the errors and deprecations). I've run the app on Android and Web so far but haven't really tested everything.
> 
> Should be do that before merging or make a thorough test before releasing the next version? @nilsreichardt

Just merging. I will fix all these bugs before the next release 👍

File: app/android/app/src/dev/google-services.json
Comment (nilsreichardt) [inline app/android/app/src/dev/google-services.json:134]: Where is this `de.codingbrain.sharezone.macos.dev` bundle id from? The macOS app has the bundle id `de.codingbrain.sharezone.app.dev`.

**Review by nilsreichardt — APPROVED** [review]
LGTM

File: app/android/app/src/dev/google-services.json
Comment (Jonas-Sander) [reply app/android/app/src/dev/google-services.json:134]: Auto-generated from the `flutterfire configure` cli.  
Since the `google-services.json` is used in Android it shouldn't matter anyways.

**Comment by nilsreichardt** [comment]
iOS build should be fixed now :)

**Comment by Jonas-Sander** [comment]
```
Downloading ios tools...                                         2,131ms
Downloading ios-profile tools...                                 1,243ms
Downloading ios-release tools...                                   911ms

Building without sound null safety ⚠️
Dart 3 will only support sound null safety, see https://dart.dev/null-safety

Archiving de.codingbrain.sharezone.app.dev...
Automatically signing iOS for device deployment using specified development team in Xcode project: AJ3PMPVZ7H
Running pod install...                                            186.3s
Running Xcode build...                                          
Xcode archive done.                                         33.7s
Failed to build iOS app
Uncategorized (Xcode): Command CompileSwiftSources failed with a nonzero exit code

Swift Compiler Error (Xcode): Concurrently-executed local function 'complete()' must be marked as '@Sendable'
/Users/builder/clone/app/ios/Pods/RevenueCat/Sources/Purchasing/Purchases/PurchasesOrchestrator.swift:961:21

Encountered error while archiving for device.


Build failed :|
Step 7 script `Build IPA` exited with status code 1
```

Seems like XCode needs to be updated: https://github.com/RevenueCat/purchases-ios/issues/2221

**Comment by nilsreichardt** [comment]
Ah, we used an old XCode version because of a Flutter bug in the 2.10.5 version. But since we're now using Flutter 3.x.x we can use the latest XCode version :)

Let's if 11f61fcb7832e988a8d7dd8d9b1df60a593a2ec9 fixed it

**Pull request merged**
